src/test/suite/scanner.test.ts

```diff
@@ -271,199 +242,12 @@ suite('ScanRunner', () => {
 
                 // ===== TEST =====
                 // Call the test method helper.
-                const args: string[] = invokeTestedMethod();
+                const args: Inputs = invokeTestedMethod();
 
                 // ===== ASSERTIONS =====
                 // Verify that the right arguments were created.
-                expect(args).to.have.lengthOf(12, 'Wrong number of args');
-                expect(args[10]).to.equal('--sfgejvmargs', 'Wrong arg');
-                expect(args[11]).to.equal(jvmArgs, 'Wrong arg');
+                expect(args['sfgejvmargs']).to.equal(jvmArgs, 'Wrong arg');
             });
         });
     });
-
-    suite('#processPathlessResults()', () => {
```

Comment:
Why did all these tests get deleted? Is this why the code coverage numbers dropped?

Reply:
That's because the methods they were testing were also deleted. Those methods were the ones parsing the stdout.
The test coverage drop is however unrelated and it's because of the noop display methods.

Reply:
Were these tests just testing implementation detail or where they testing behavior? Ideally tests should lock in requirements and contracts... not implementation detail. So if the requirement is that we somehow handle errors/warnings a specific way - then we should keep that tested somehow.

I don't know exactly what these tests were for but they see things like... 
"Returns violations found during successful scan"
"Escalates method-not-found warning to error"
Are these no longer a part of the overall contract?

If they are, then tests should be modified instead of deleted (ideally to not be implementation specific if possible).

If they were just testing implementation detail... then fine these can go away if they truly didn't lock in some requirement or contract.